server/chat-plugins/helptickets-auto.ts

```diff
@@ -0,0 +1,670 @@
+			`(${result.reason || `report from ${ticket.creator}`})`
+		);
+		globalModlog(
+			'WARN', user, result.reason || `report from ${ticket.creator}`
+		);
+		return `${user} was automatically warned. Thank you for reporting.`;
+	},
+};
+
+export async function getMessageAverages(messages: string[]) {
+	const counts: Record<string, {count: number, raw: number}> = {};
+	const classified = [];
+	for (const message of messages) {
+		const res = await classifier.classify(message);
+		if (!res) continue;
+		classified.push(res);
+		for (const k in res) {
+			if (!counts[k]) counts[k] = {count: 0, raw: 0};
+			counts[k].count++;
+			counts[k].raw += res[k];
+		}
+	}
+	const averages: Record<string, number> = {};
+	for (const k in counts) {
+		averages[k] = counts[k].raw / counts[k].count;
+	}
+	return {averages, classified};
+}
+
+interface CheckerResult {
+	action: string;
+	user: User | ID;
+	result: Record<string, number>;
+	reason: string;
+	roomid?: string;
+	displayReason?: string;
+	proof?: string;
+}
+
+type CheckerOutput = void | Map<string, CheckerResult>;
+
+export const checkers: {
+	[k: string]: (ticket: TicketState & {text: [string, string]}) => CheckerOutput | Promise<CheckerOutput>,
+} = {
+	async inapname(ticket) {
+		const id = toID(ticket.text[0]);
+		const user = Users.getExact(id);
+		if (user && !user.trusted) {
+			const result = await classifier.classify(user.name);
+			if (!result) return;
+			const keys = ['identity_attack', 'sexual_explicit', 'severe_toxicity'];
+			const matched = keys.some(k => result[k] >= 0.4);
+			if (matched) {
+				const modlog = await getModlog({
+					ip: user.latestIp,
+					actions: ['FORCERENAME', 'NAMELOCK', 'WEEKNAMELOCK'],
+				});
+				let action = determinePunishment('inapname', result, modlog);
+				if (!action) action = 'forcerename';
+				return new Map([[user.id, {
+					action,
+					user,
+					result,
+					reason: "Username detected to be breaking username rules",
+				}]]);
+			}
+		}
+	},
+	async inappokemon(ticket) {
+		const actions = new Map();
+		const links = [...getBattleLinks(ticket.text[0]), ...getBattleLinks(ticket.text[1])];
+		for (const link of links) {
+			const log = await getBattleLog(link);
+			if (!log) continue;
+			for (const [userid, pokemon] of Object.entries(log.pokemon)) {
+				let result: {
+					action: string, name: string, result: Record<string, number>, replay: string,
+				} | null = null;
+				for (const set of pokemon) {
+					if (!set.name) continue;
+					const results = await classifier.classify(set.name);
+					if (!results) continue;
+					// atm don't factor in modlog
+					const curAction = determinePunishment('inappokemon', results, []);
+					if (curAction && (!result || supersedes(curAction, result.action))) {
+						result = {action: curAction, name: set.name, result: results, replay: link};
+					}
+				}
+				if (result) {
+					// if action exists, the rest will exist. this is just for TS
+					actions.set(toID(userid), {
```

Comment:
`userid` implies it's an ID... maybe name it `user` or `username` if it still needs to be converted to an ID?

Reply:
Fair.